framework/basehandlers.py

```diff
@@ -153,21 +153,22 @@ def delete(self, *args, **kwargs):
     handler_data = self.do_delete(*args, **kwargs)
     return flask.jsonify(handler_data), headers
 
-  def do_get(self):
+  def do_get(self, **kwargs):
     """Subclasses should implement this method to handle a GET request."""
+    # Every API handler must handle GET.
     raise NotImplementedError()
 
-  def do_post(self):
+  def do_post(self, **kwargs):
     """Subclasses should implement this method to handle a POST request."""
-    raise NotImplementedError()
+    self.abort(400, msg='Unexpected HTTP method')
```

Comment:
I had to remind myself of the various HTTP status codes. Looking at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses), it seems like a 405 would be more appropriate than 400. Although, using 405 seems to require also responding with an [Allow header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow).

That said, if this is to primarily handle malicious requests, then maybe it doesn't matter if we use the "best" status code.

Reply:
You are right that 405 is a better response status.  I went ahead and used that.  Flask already has some support for setting the allow header, I just needed to tell it which methods are defined in the API handler subclass.